File: frontend/src/app/model/Propos.ts
export class Propos{
    propo_numero: number ;
    propo_date: Date ;
    propo_codeermediaire: number ; 
    propo_codecompagnie: String ;
    propo_numerobranche: number ; // cf table branche
    propo_numerocategorie: number ; // cf table categorie
    propo_numerosouscripteur: number ;
    propo_numeroprospect: number;
	propo_dateeffet1er: Date;
	propo_dateannivcontrat: Date;
	propo_dateeffet: Date;
	propo_dateecheance: Date;
	propo_dureecontrat: number;
	propo_typecontrat: number;
	propo_typegestion: number;
	propo_codefractionnemen: number;
	propo_mtnprimenetref: number;
	propo_mtnprimenettot: number;
	propo_accesoirecompagnie: number;
	propo_accessoireapporteur: number;
	propo_taxe: number;
	propo_commission: number;
	propo_mtnprimebrut: number;
	propo_coefbonus: number;
	propo_coefremisecommerciale: number;
	propo_codeproduit: number;
	propo_datesituationproposition: Date;
	propo_statusproposition: String ;
	propo_exontaxeenr: String ;
	propo_codetaxeenr: number;
	propo_exontva: String ;
	propo_codetva: number;
	propo_exontps: String;
	propo_codetps: number;
	propo_dateexon: Date;
	propo_codeutil: number;
	propo_datemaj: Date;
	propo_datetransformationcontrat: Date;
}

File: api/src/main/java/sn/jmad/sonac/model/Propos.java
package sn.jmad.sonac.model;

import java.io.Serializable;
import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import org.hibernate.annotations.Generated;
import org.hibernate.annotations.GenerationTime;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "propos")
@Data
@AllArgsConstructor

public class Propos implements Serializable{
	@Column(columnDefinition = "serial")
	@Generated(GenerationTime.INSERT)
	private Long propo_id;
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long propo_numero;
	private Date propo_date;
	private Long propo_codeintermediaire;
	private String propo_codecompagnie;
	private Long propo_numerobranche;
	private Long propo_numerocategorie;
	private Long propo_numerosouscripteur;
	private Long propo_numeroprospect;
	private Date propo_dateeffet1er;
	private Date propo_dateannivcontrat;
	private Date propo_dateeffet;
	private Date propo_dateecheance;
	private int propo_dureecontrat;
	private int propo_typecontrat;
	private int propo_typegestion;
	private int propo_codefractionnement;
	private Long propo_mtnprimenetref;
	private Long propo_mtnprimenettot;
	private Long propo_accesoirecompagnie;
	private Long propo_accessoireapporteur;
	private Long propo_taxe;
	private Long propo_commission;
	private Long propo_mtnprimebrut;
	private Long propo_coefbonus;
	private Long propo_coefremisecommerciale;
	private Long propo_codeproduit;
	private Date propo_datesituationproposition;
	private String propo_statusproposition;
	private String propo_exontaxeenr;
	private int propo_codetaxeenr;
	private String propo_exontva;
	private int propo_codetva;
	private String propo_exontps;
	private int propo_codetps;
	private Date propo_dateexon;
	private String propo_codeutil;
	private Date propo_datemaj;
	private Date propo_datetransformationcontrat;
	private int active;
	
	public Propos() {
		super();
	}
	
	
}
